lib/node_modules/@stdlib/ndarray/some-by/docs/types/index.d.ts

```diff
@@ -0,0 +1,344 @@
+* @returns boolean indicating whether an ndarray element passes a test
+*/
+type Unary<T, ThisArg> = ( this: ThisArg, value: T ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param indices - current array element indices
+* @returns boolean indicating whether an ndarray element passes a test
+*/
+type Binary<T, ThisArg> = ( this: ThisArg, value: T, indices: Array<number> ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param indices - current array element indices
+* @param arr - input array
+* @returns boolean indicating whether an ndarray element passes a test
+*/
+type Ternary<T, ThisArg> = ( this: ThisArg, value: T, indices: Array<number>, arr: typedndarray<T> ) => boolean;
+
+/**
+* Returns a boolean indicating whether an element passes a test.
+*
+* @param value - current array element
+* @param indices - current array element indices
+* @param arr - input array
+* @returns boolean indicating whether an ndarray element passes a test
+*/
+type Predicate<T, ThisArg> = Nullary<ThisArg> | Unary<T, ThisArg> | Binary<T, ThisArg> | Ternary<T, ThisArg>;
+
+/**
+* Base options.
+*/
+interface BaseOptions {
+	/**
+	* List of dimensions over which to perform the reduction.
+	*/
+	dims?: ArrayLike<number>;
+}
+
+/**
+* Options.
+*/
+interface Options extends BaseOptions {
+	/**
+	* Boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions. Default: `false`.
+	*/
+	keepdims?: boolean;
+}
+
+/**
+* Interface describing `someBy`.
+*/
+interface SomeBy {
+	/**
+	* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function along one or more dimensions.
+	*
+	* @param x - input ndarray
+	* @param n - number of elements which must pass a test
+	* @param predicate - predicate function
+	* @param thisArg - predicate execution context
+	* @returns output ndarray
+	*
+	* @example
+	* var Float64Array = require( '@stdlib/array/float64' );
+	* var isEven = require( '@stdlib/assert/is-even' ).isPrimitive;
+	* var ndarray = require( '@stdlib/ndarray/ctor' );
+	*
+	* // Create a data buffer:
+	* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] );
+	*
+	* // Define the shape of the input array:
+	* var sh = [ 3, 1, 2 ];
+	*
+	* // Define the array strides:
+	* var sx = [ 4, 4, 1 ];
+	*
+	* // Define the index offset:
+	* var ox = 1;
+	*
+	* // Create an input ndarray:
+	* var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+	*
+	* // Perform reduction:
+	* var out = someBy( x, 3, isEven );
+	* // returns <ndarray>
+	*
+	* var v = out.get();
+	* // returns true
+	*/
+	<T = unknown, U = unknown>( x: ndarray, n: ndarray | unknown, predicate: Predicate<T,U>, thisArg?: ThisParameterType<Predicate<T, U>> ): boolndarray;
```

Comment:
We can do better here.

```suggestion
	<T = unknown, U = unknown>( x: ndarray, n: integerndarray | number, predicate: Predicate<T,U>, thisArg?: ThisParameterType<Predicate<T, U>> ): boolndarray;
```

Applies here and below.